### PR TITLE
Add: Taiko L2 Blockchain and TAIKO Token

### PR DESCRIPTION
**Pull request by JBScaled**
I am requesting to add the Taiko blockchain, an Ethereum L2 scaling solution, to Trust Wallet as a blockchain that can also support ERC20 tokens. I am also requesting to add the Taiko Token as an ERC20 token that exists on both Ethereum Mainnet and Taiko L2 Mainnet. 

The Layer 2 network was launched nearly two weeks ago and has since become permissionless by opening up the ability to prove and sequence blocks to the community. 

[Launch coverage](https://www.theblock.co/post/296831/ethereum-l2-taiko-mainnet)
[Permissionless coverage](https://www.theblock.co/post/298877/taiko-enables-permissionless-sequencing-and-proving-after-mainnet-rollout)

I understand that there is a requirement to have 10,000 token holders before a token can be considered to be added to Trust Wallet. Please see [here](https://taikoscan.io/token/0xa9d23408b9ba935c230493c40c73824df71a0975) for reference that on Taiko Mainnet we now have 130,000+ holders and have processed 640,000+ transfers of that token. On [Ethereum Mainnet](https://etherscan.io/token/0x10dea67478c5f8c5e2d90e5e9b26dbe60c54d800) we haven't reached the 10,000 holders yet, but the token has already been transacted over 80,000 times. We would appreciate it if the unique holders on Taiko L2 Mainnet would be considered when evaluating the Ethereum token as well. 

It is also worth mentioning that a large number of our early holder did receive an airdrop, and this airdrop was earned by participating in our 6 testnet phases, and the tokens needed to be manually claimed by the wallets through the [webpage here](https://claim.taiko.xyz/). Information on the Airdrop can be [found here](https://taiko.mirror.xyz/VSOtILX2DQsc_6IMt5hBT1fEYSH8243pZ8IA_pBfHks).

Lastly here is our [Tokenomics whitepaper](https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/docs/tokenomics-whitepaper.pdf) and our [Protocol Audits](https://github.com/taikoxyz/taiko-mono/tree/main/packages/protocol/audit).

### CODE REVIEW

File: blockchains/ethereum/assets/0x10dea67478c5F8C5E2D90e5E9B26dBe60c54d800/info.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            "name": "whitepaper",
+            "url": "https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/docs/tokenomics-whitepaper.pdf"
+        },
+        {
+            "name": "blog",
+            "url": "https://taiko.mirror.xyz/"
+        },
+        {
+            "name": "discord",
+            "url": "https://discord.gg/taikoxyz"
```
Comment (dantaik) [inline blockchains/ethereum/assets/0x10dea67478c5F8C5E2D90e5E9B26dBe60c54d800/info.json:37]:
```suggestion
            "url": "https://discord.com/invite/taikoxyz"
```

**Comment by dantaik** [comment]
Regarding: low token circulation: number of holders is 109, below limit of 10000

TAIKO token lives on two networks, Ethereum and Taiko. The one on Ethereum (0x10dea67478c5f8c5e2d90e5e9b26dbe60c54d800) used to have 100% supply, but we will gradually bridge over these tokens to Taiko Mainnet (0xa9d23408b9ba935c230493c40c73824df71a0975). As of now we have 141 holders on Ethereum and 121,275 tokens on Taiko.

**Comment by dantaik** [comment]
> @JBScaled, kind reminder: please pay fee: **500 TWT** to the address `0xD795928cB935043653ee7060d1CC768029a098f4`. Alternatively, 2.5 BNB is also accepted (same address).
> 
> **QR** code: [Trust](https://api.qrserver.com/v1/create-qr-code/?size=200x200&data=https%3A%2F%2Flink.trustwallet.com%2Fsend%3Fasset%3Dc20000714_t0x4B0F1812e5Df2A09796481Ff14017e6005508003%26address%3D0xD795928cB935043653ee7060d1CC768029a098f4%26amount%3D500) | [other wallet](https://api.qrserver.com/v1/create-qr-code/?size=200x200&data=0xD795928cB935043653ee7060d1CC768029a098f4%3Famount%3D500) See the [Pull Request Fee FAQ](https://developer.trustwallet.com/assets/faq).

https://bscscan.com/tx/0x32efdf9b6f63047f8034fda4c4962bdb0c89fa734165bf25eafd8e1b5f5e8d12

**Comment by JBScaled** [comment]
Can we get some human input here please @Cryptocool1  @cryptomanz1

**Comment by Cryptocool1** [comment]
• What is the purpose of this token? How will it benefit Trust Wallet users?
 • How is the token being distributed? Are you conducting an airdrop?
 • Did you sold the tokens at an Initial Coin Offering (ICO) or a token presale?
 • Is the token listed on a centralized/decentralized exchange?
 • Is a Smart Contract audit completed? Where can we find this information?

**Comment by CurlyBracketEffect** [comment]
Hey @Cryptocool1 thanks for your reply. Here are the answers to your questions:

> What is the purpose of this token? How will it benefit Trust Wallet users?

The Taiko token is the utility token of the Taiko L2 network. Here are a few of its main utilities: 

1. It is used to reward network participants for proposing new valid L2 blocks to be proven and included in Ethereum L1 blocks. 
2. It is used as the validity and contestation bonds in our Based Contestable Rollup design as described in [this article](https://taiko.mirror.xyz/Z4I5ZhreGkyfdaL5I9P0Rj0DNX4zaWFmcws-0CVMJ2A)
3. It is the governance token for the Taiko protocol. As the network matures all token holders will be able to participate in the decision-making process through DAO governance. Taiko tokens will give holders voting power in the DAO. 

Adding Taiko to Trust Wallet would benefit your users by giving them access to a new Ethereum Layer 2 scaling solution that has been [endorsed by Vitalk Buterin](https://www.binance.com/en/square/post/2024-05-25-ethereum-co-founder-vitalik-buterin-proposes-first-block-on-taiko-mainnet-8568889837921). 

> How is the token being distributed? Are you conducting an airdrop?
> Did you sold the tokens at an Initial Coin Offering (ICO) or a token presale?

The initial circulating supply was distributed to the community that supported its development throughout the 6 pre-mainnet phases. This included retail testnet participants, but also people that contributed to core GitHub repos. These tokens were not dropped into each person's wallet, but rather they needed to claim them manually after passing through a stringent Sybil detection process. Additionally, there was an early token sale offered by ByBit. Together this amounted to less than 10% of the token supply being released. Moving forward we have an additional 10% of Tokens earmarked for our early adopter program called [Trailblazers](https://taiko.mirror.xyz/589CthdX0Qw-hRBeZarEQpjck-v4lJ-gtKz3SZvovt8) which is planned to run for the next 4 years. We also have a large allocation of tokens set aside to fund our community grants program. Beyond that, allocations for the team and investors will be distributed under a multiyear vesting schedule. 

For a deeper dive into the Taiko Token, you can view our [Tokenomics Whitepaper](https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/docs/tokenomics-whitepaper.pdf).

> Is the token listed on a centralized/decentralized exchange?

Yes, the Taiko Token has already been listed on several centralized and decentralized exchanges, such as Crypto.com, Bybit, Bitget, MEXC, Kucoin, Uniswap, Pancakeswap, and iZiSwap. It is also listed on the notable token metric sites [Coingecko](https://www.coingecko.com/en/coins/taiko) and [Coinmatketcap](https://coinmarketcap.com/currencies/taiko/).

> Is a Smart Contract audit completed? Where can we find this information?

Yes, I included a link to the [protocol audits](https://github.com/taikoxyz/taiko-mono/tree/main/packages/protocol/audit) as one of the additional links to our blockchain listing. These audits were performed by three separate reputable teams. 

Thank you for the questions, I hope I have answered each adequately. I look forward to hearing what the next steps are.

**Comment by CurlyBracketEffect** [comment]
Hey @Cryptocool1 thanks for your reply. Here are the answers to your questions:

> What is the purpose of this token? How will it benefit Trust Wallet users?

The Taiko token is the utility token of the Taiko L2 network. Here are a few of its main utilities:
1. It is used to reward network participants for proposing new valid L2 blocks to be proven and included in Ethereum L1 blocks.
2. It is used as the validity and contestation bonds in our Based Contestable Rollup design as described in [this article](https://taiko.mirror.xyz/Z4I5ZhreGkyfdaL5I9P0Rj0DNX4zaWFmcws-0CVMJ2A)
3. It is the governance token for the Taiko protocol. As the network matures all token holders will be able to participate in the decision-making process through DAO governance. Taiko tokens will give holders voting power in the DAO.
Adding Taiko to Trust Wallet would benefit your users by giving them access to a new Ethereum Layer 2 scaling solution that has been [endorsed by Vitalk Buterin](https://www.binance.com/en/square/post/2024-05-25-ethereum-co-founder-vitalik-buterin-proposes-first-block-on-taiko-mainnet-8568889837921).

> How is the token being distributed? Are you conducting an airdrop? 
> Did you sold the tokens at an Initial Coin Offering (ICO) or a token presale?

The initial circulating supply was distributed to the community that supported its development throughout the 6 pre-mainnet phases. This included retail testnet participants, but also people that contributed to core GitHub repos. These tokens were not dropped into each person's wallet, but rather they needed to claim them manually after passing through a stringent Sybil detection process. Additionally, there was an early token sale offered by ByBit. Together this amounted to less than 10% of the token supply being released. Moving forward we have an additional 10% of Tokens earmarked for our early adopter program called [Trailblazers](https://taiko.mirror.xyz/589CthdX0Qw-hRBeZarEQpjck-v4lJ-gtKz3SZvovt8) which is planned to run for the next 4 years. We also have a large allocation of tokens set aside to fund our community grants program. Beyond that, allocations for the team and investors will be distributed under a multiyear vesting schedule.

For a deeper dive into the Taiko Token, you can view our [Tokenomics Whitepaper](https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/docs/tokenomics-whitepaper.pdf).

> Is the token listed on a centralized/decentralized exchange?

Yes, the Taiko Token has already been listed on several centralized and decentralized exchanges, such as Crypto.com, Bybit, Bitget, MEXC, Kucoin, Uniswap, Pancakeswap, and iZiSwap. It is also listed on the notable token metric sites [Coingecko](https://www.coingecko.com/en/coins/taiko) and [Coinmatketcap](https://coinmarketcap.com/currencies/taiko/).

> Is a Smart Contract audit completed? Where can we find this information?

Yes, I included a link to the [protocol audits](https://github.com/taikoxyz/taiko-mono/tree/main/packages/protocol/audit) as one of the additional links to our blockchain listing. These audits were performed by three separate reputable teams.

Thank you for the questions, I hope I have answered each adequately. I look forward to hearing what the next steps are.

**Comment by CurlyBracketEffect** [comment]
Hi @Cryptocool1 I have taken the time to write out detailed answers to your questions along with supporting links, but every time I post the reply, the merge-fee-bot is deleting it automatically. Can you please advise me on what is causing this so that I can answer your questions?

**Comment by CurlyBracketEffect** [comment]
> • What is the purpose of this token? How will it benefit Trust Wallet users? • How is the token being distributed? Are you conducting an airdrop? • Did you sold the tokens at an Initial Coin Offering (ICO) or a token presale? • Is the token listed on a centralized/decentralized exchange? • Is a Smart Contract audit completed? Where can we find this information?

I have tried to reply to these questions several times now, and every time the bot deletes my comment. What am I supposed to do?

**Comment by CurlyBracketEffect** [comment]
I am trying to answer, but my comments keep getting deleted

**Review by Abeanna — COMMENTED** [review]
Good

**Comment by JBScaled** [comment]
Hey [@Cryptocool1](https://github.com/Cryptocool1) thanks for your reply. Here are the answers to your questions:

> What is the purpose of this token? How will it benefit Trust Wallet users?

The Taiko token is the utility token of the Taiko L2 network. Here are a few of its main utilities:
1. It is used to reward network participants for proposing new valid L2 blocks to be proven and included in Ethereum L1 blocks.
2. It is used as the validity and contestation bonds in our Based Contestable Rollup design as described in [this article](https://taiko.mirror.xyz/Z4I5ZhreGkyfdaL5I9P0Rj0DNX4zaWFmcws-0CVMJ2A)
3. It is the governance token for the Taiko protocol. As the network matures all token holders will be able to participate in the decision-making process through DAO governance. Taiko tokens will give holders voting power in the DAO.

Adding Taiko to Trust Wallet would benefit your users by giving them access to a new Ethereum Layer 2 scaling solution that has been [endorsed by Vitalk Buterin](https://www.binance.com/en/square/post/2024-05-25-ethereum-co-founder-vitalik-buterin-proposes-first-block-on-taiko-mainnet-8568889837921).

> How is the token being distributed? Are you conducting an airdrop? 
> Did you sold the tokens at an Initial Coin Offering (ICO) or a token presale?

The initial circulating supply was distributed to the community that supported its development throughout the 6 pre-mainnet phases. This included retail testnet participants, but also people that contributed to core GitHub repos. These tokens were not dropped into each person's wallet, but rather they needed to claim them manually after passing through a stringent Sybil detection process. Additionally, there was an early token sale offered by ByBit. Together this amounted to less than 10% of the token supply being released. Moving forward we have an additional 10% of Tokens earmarked for our early adopter program called [Trailblazers](https://taiko.mirror.xyz/589CthdX0Qw-hRBeZarEQpjck-v4lJ-gtKz3SZvovt8) which is planned to run for the next 4 years. We also have a large allocation of tokens set aside to fund our community grants program. Beyond that, allocations for the team and investors will be distributed under a multiyear vesting schedule.

For a deeper dive into the Taiko Token, you can view our [Tokenomics Whitepaper](https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/docs/tokenomics-whitepaper.pdf).

> Is the token listed on a centralized/decentralized exchange?

Yes, the Taiko Token has already been listed on several centralized and decentralized exchanges, such as Crypto.com, Bybit, Bitget, MEXC, Kucoin, Uniswap, Pancakeswap, and iZiSwap. It is also listed on the notable token metric sites [Coingecko](https://www.coingecko.com/en/coins/taiko) and [Coinmarketcap](https://coinmarketcap.com/currencies/taiko/).

> Is a Smart Contract audit completed? Where can we find this information?

Yes, I included a link to the [protocol audits](https://github.com/taikoxyz/taiko-mono/tree/main/packages/protocol/audit) as one of the additional links to our blockchain listing. These audits were performed by three separate reputable teams.

Thanks again for the questions, I hope I have answered each adequately. I look forward to hearing what the next steps are.

**Comment by dantaik** [comment]
@Cryptocool1 @Abeanna is it possible to know a rough ETA for this request? Thank you.

**Comment by JBScaled** [comment]
Hey @Cryptocool1 I just wanted to check in on this PR. Can you provide some clarity on the process here, please?

**Comment by JBScaled** [comment]
@Cryptocool1 it has been 2 weeks now, can you please provide some feedback on what the next steps are?

**Comment by cryptomanz1** [comment]
At the moment we do not merge new blockchains into WalletCore without a positive decision from our Business Development team.
Please create a new ticket, a request to add the blockchain to Trust Wallet:
https://support.trustwallet.com/en/support/tickets/new

**Comment by Cryptocool1** [comment]
For now we can add only your ERC20 token

**Pull request merged**